### ISSUE TITLE
Don't show unused arguments for first argument when the second is used

### ISSUE THREAD

**Issue by tandrewnichols** [syngan/vim-vimlint#112]
If a function takes two arguments and the second one _is_ used, the first shouldn't be flagged as unused because the alternative way of writing that using something like `a:2` is less clear. The reason "Just refactor the function to take one argument" is invalid is because some built in functions pass multiple parameters. E.g.

```vim
function! MapFunction(index, item)
  echo a:item
endfunction

map(list, function('MapFunction'))
```

This complains that `a:index` is unused, but you can't refactor `MapFunction` to take only one argument because that's not how `map()` works with a funcref.

---

Semi-related: it would be nice to just turn off undef argument warnings for funcrefs passed to built-in functions because things like `timer_start` pass an argument to the funcref you provide, and you can't just not accept it or vim complains that there are too many arguments. The best you can do is:

```vim
function! Delayed(...)
  " Why would you ever need the timer id of the timer that _just_ completed???
endfunction

timer_start(200, function('Delayed'))
```

which is an okay work around, and since I expect this secondary issue would be pretty difficult to solve, I imagine it will have to suffice. Just thought it was worth mentioning as long as I was already opening an issue.
